src/util/Spawner.java

```diff
@@ -0,0 +1,104 @@
+package util;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+import rts.*;
+import rts.units.*;
+
+public class Spawner {
+    String mode;
+
+    static UnitTypeTable utt;
+
+    UnitType resourceType;
+    UnitType baseType;
+    UnitType barracksType;
+    UnitType workerType;
+    UnitType lightType;
+    UnitType heavyType;
+    UnitType rangedType;
+
+    ArrayList<UnitType> types;
+
+    public Spawner(String mode, UnitTypeTable a_utt) {
+        this.mode = mode;
+        utt = a_utt;
+        types = new ArrayList<UnitType>();
+        resourceType = utt.getUnitType("Resource");
+        baseType = utt.getUnitType("Base");
+        barracksType = utt.getUnitType("Barracks");
+        workerType = utt.getUnitType("Worker");
+        lightType = utt.getUnitType("Light");
+        heavyType = utt.getUnitType("Heavy");
+        rangedType = utt.getUnitType("Ranged");
+
+        // types.add(workerType);
+        types.add(lightType);
+        types.add(heavyType);
+        types.add(rangedType);
+    }
+
+    public void spawn(GameState gs) {
+        switch (mode) {
+            case "blockade":

```

Comment:
is there any place I can find the documentation for each of these modes?

Reply:
I am going to write one up, I forgot I only explained things in presentations, but never formally wrote documentation.